VPLAY-11239  [Xione UK][VIPA EPG] GLib-GObject with RialtoMSESubtitleSink Warning Errors in Console When Tuning to Linear Channels

pstroffolino opened the pull request:
[VPLAY-11239 ](https://ccp.sys.comcast.net/browse/VPLAY-11239) [Xione UK][VIPA EPG] GLib-GObject with RialtoMSESubtitleSink Warning Errors in Console When Tuning to Linear Channels

Reason for Change: simplifications & fixes in InterfacePlayerRDK.  Removes unnecessary use of isRialto parameter, and avoid use of rialtomsesubtitlesink as an audio sink, which spews warnings

Test Guidance: general testing of VIPA and non-VIPA playback; confirm runtime gstreamer warnings cited in ticket no longer seen

Risk: Medium

varshnie committed:
VPLAY-10999:[AAMP] Profile playback stop operation

Reason for change:fragmentcollectorthread optimization
Test Procedure: Refer jira ticket
Priority: P1

Signed-off-by: varshnie <varshniblue14@gmail.com>

pstroffolino committed:
Reason for Change: minor cleanup

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: fixed comment

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-11239  [Xione UK][VIPA EPG] GLib-GObject with RialtoMSESubtitleSink Warning Errors in Console When Tuning to Linear Channels

Reason for Change: simplify InterfacePlayerRDK abstraction and avoid using text sink as audio sink

Test Guidance: general regression (VIPA and non-VIPA) and

No more warnings like:
4246:2025-09-26T10:06:28.121Z com.sky.as.apps_com.comcast.viper_ipa[3854]:  <stderr> (WPEWebProcess:33): GLib-GObject-WARNING **: ../glib-2.74.6/nal 'first-audio-frame-callback' is invalid for instance '0x8e90d090' of type 'RialtoMSESubtitleSink'

Risk: Medium

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-11239

pstroffolino committed:
Reason for Change: add'l cleanup

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: simplifications for DefaultSocInterface::IsPlatformSegmentReady, DefaultSocInterface::IsVideoMaster

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino commented on middleware/InterfacePlayerRDK.cpp:
```diff
@@ -1775,11 +1775,10 @@ void InterfacePlayerRDK::InitializeSourceForPlayer(void *PlayerInstance, void *
-		if ((privatePlayer->gstPrivateContext->usingRialtoSink) &&
-			(privatePlayer->socInterface->IsPlatformSegmentReady(privatePlayer->gstPrivateContext->video_sink, privatePlayer->gstPrivateContext->usingRialtoSink)))
+		
+		if( privatePlayer->gstPrivateContext->usingRialtoSink )
```
this is functional change - is it ok?  when exactly do we need to set handle-segment-change property?  What does it do?  Is it for use always/only when using Rialto?  

IsPlatformSegmentReady previously was associated with use of video master - is that correct?

pstroffolino committed:
Reason for Change: simplifications - more Rialto-specific code pushed into default SocInterface

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: avoid functional change for handle-segment-change trigger

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: updates to FakeSocInterface.cpp

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

jfagunde commented on middleware/vendor/SocInterface.h:
```diff
@@ -456,26 +450,13 @@ class SocInterface
 	 */
-	virtual bool IsVideoMaster(GstElement *videoSink, bool isRialto){return true;}
-
+	virtual bool IsVideoMaster(GstElement *videoSink) = 0; // {return true;}
```
I'd delete the commented out implementation.

jfagunde commented on middleware/vendor/default/DefaultSocInterface.h:
```diff
@@ -29,168 +29,153 @@
+	 */
+	void SetHevcCaps(GstCaps *caps)override;
+	
+	bool IsVideoMaster(GstElement *videoSink)override{
```
I'd move this implementation to the source file DefaultSocInterface.cpp.

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-11239

pstroffolino committed:
Reason for Change: updated middleware fakes

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Reason for Change: quick patch for utests

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Update SocInterface.h

pstroffolino committed:
Update DefaultSocInterface.h

pstroffolino committed:
Update DefaultSocInterface.cpp

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-11239

pstroffolino committed:
Reason for Change: fake fix

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-11239

rekhap2kandhavelan committed:
Enhance PauseOnPlaybackTests with privatePlayer setup

Updated PauseOnPlaybackTests to include privatePlayer initialization and modified expectations for frame-step-on-preroll property.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=585) <br/>All committers have signed the CLA.

rekhap2kandhavelan committed:
Enhance video sink and decoder checks with additional sinks

pstroffolino committed:
Merge branch 'dev_sprint_25_2' into feature/VPLAY-11239
